src/checkbox/index.tsx

```diff
@@ -0,0 +1,141 @@
+import { ThemedProperties } from '@dojo/framework/core/mixins/Themed';
+import { FocusProperties } from '@dojo/framework/core/mixins/Focus';
+import Label from '../label/index';
+import { formatAriaProperties } from '../common/util';
+import { uuid } from '@dojo/framework/core/util';
+
+export interface CheckboxProperties extends ThemedProperties, FocusProperties {
+	/** Custom aria attributes */
+	aria?: { [key: string]: string | null };
+	/**  Checked/unchecked property of the radio */
+	checked?: boolean;
+	disabled?: boolean;
+	label?: string;
+	labelAfter?: boolean;
+	labelHidden?: boolean;
+	name?: string;
+	onBlur?(): void;
+	onFocus?(): void;
+	/** Label to show in the "on" positin of a toggle */
+	onLabel?: DNode;
+	onOut?(): void;
+	onOver?(): void;
+	onValue?(checked: boolean): void;
+	readOnly?: boolean;
+	required?: boolean;
+	valid?: boolean;
+	/** The current value */
+	value?: string;
+	widgetId?: string;
+}
+
+const factory = create({ coreTheme, focus }).properties<CheckboxProperties>();
+
+export const Checkbox = factory(function Checkbox({
+	properties,
+	middleware: { coreTheme, focus }
+}) {
+	const _uuid = uuid();
+	const {
+		aria = {},
+		checked = false,
+		classes,
+		disabled,
+		label,
+		labelAfter = true,
+		labelHidden,
+		name,
+		onBlur,
+		onFocus,
+		onValue,
+		onOut,
+		onOver,
+		readOnly,
+		required,
+		theme,
+		valid,
+		widgetId = _uuid
+	} = properties();
+
+	const themeCss = coreTheme.classes(css);
+
+	const getRootClasses = () => [
+		themeCss.root,
+		checked ? themeCss.checked : null,
+		disabled ? themeCss.disabled : null,
+		focus.isFocused('root') ? themeCss.focused : null,
+		valid === false ? themeCss.invalid : null,
+		valid === true ? themeCss.valid : null,
+		readOnly ? themeCss.readonly : null,
+		required ? themeCss.required : null
+	];
+
+	const children = [
+		<div classes={themeCss.inputWrapper}>
+			<input
+				id={widgetId}
+				{...formatAriaProperties(aria)}
+				classes={themeCss.input}
+				checked={checked}
+				disabled={disabled}
+				focus={focus.shouldFocus()}
+				aria-invalid={valid === false ? 'true' : null}
+				name={name}
+				readonly={readOnly}
+				aria-readonly={readOnly === true ? 'true' : null}
+				required={required}
+				type="checkbox"
+				onblur={() => onBlur && onBlur()}
+				onchange={(event: Event) => {
+					event.stopPropagation();
+					const checkbox = event.target as HTMLInputElement;
+					onValue && onValue(checkbox.checked);
+				}}
+				onfocus={() => onFocus && onFocus()}
+				onpointerenter={() => onOver && onOver()}
+				onpointerleave={() => onOut && onOut()}
+			/>
+			<div classes={themeCss.background}>
+				<svg classes={themeCss.checkmark} viewBox="0 0 24 24">
+					<path
+						classes={themeCss.checkmarkPath}
+						fill="none"
+						d="M1.73,12.91 8.1,19.28 22.79,4.59"
+					/>
+				</svg>
+				<div classes={themeCss.mixedMark} />
+			</div>
+			<div classes={themeCss.ripple} />
```

Comment:
i really don't think we should be adding things like `ripple` classes / dom nodes

Reply:
@tomdye I don't like it either, particularly the svg element that's really only used for the material theme. Do you have a suggestion about how best to handle this?

Reply:
we don't implement ripple elsewhere so i'd suggest you just leave it out

Reply:
I can remove it. Without the ripple there won't be a subtle hover effect on the checkbox.

**With ripple**
![with-classes](https://user-images.githubusercontent.com/293805/68825155-c335d900-065e-11ea-943d-54f5f954229e.gif)
There is a subtle hover highlight that appears

**Without ripple**
![without-classes](https://user-images.githubusercontent.com/293805/68825180-d5b01280-065e-11ea-8eac-7f0015abafac.gif)
The hover is no longer visible.